Python/symtable.c

```diff
@@ -803,6 +818,10 @@ inline_comprehension(PySTEntryObject *ste, PySTEntryObject *comp,
             }
         }
     }
+    comp->ste_free = PySet_Size(comp_free) > 0;
+    if (remove_dunder_class && PyDict_DelItem(comp->ste_symbols, k) < 0) {
```

Comment:
Isn't `k` here whatever symbol was last yielded by the iterator? If there is another name after `__class__`, we could delete the wrong name.

Reply:
Oops, thank you :) Moved this out of the loop (since it's not safe to modify the dict during iteration) and failed to update it. Fixed.